CONTRIBUTING.md

```diff
@@ -59,7 +59,7 @@ git clone --recurse-submodules https://github.com/determined-ai/determined.git
 - NPM (>= 8)
 - Docker (>= 19.03)
 - Helm (>= 3.0.0)
-- protobuf-compiler (>= 3.0)
+- protobuf-compiler (>= 3.15)
```

Comment:
`--experimental_allow_proto3_optional` was allowed in `>= 3.12` but this is fine to me

Reply:
https://github.com/protocolbuffers/protobuf/blob/main/CHANGES.txt#L1076

Reply:
> Singular, non-message typed fields in proto3 now support
    presence tracking.

lol, possibly the most complicated way to describe the feature